third_party/amd/lib/Dialect/TritonAMDGPU/IR/Dialect.cpp

```diff
@@ -447,6 +447,45 @@ LogicalResult ConcatOp::verify() {
   return success();
 }
 
+LogicalResult LocalLoadPackedTransposedOp::verify() {
+  auto srcTy = getSrc().getType();
+  auto dstTy = getType();
+  auto srcShape = srcTy.getShape();
+
+  auto dotEnc = cast<DotOperandEncodingAttr>(dstTy.getEncoding());
+  auto sharedEnc =
+      cast<triton::gpu::SwizzledSharedEncodingAttr>(srcTy.getEncoding());
+  auto order = sharedEnc.getOrder();
+  bool isA = dotEnc.getOpIdx() == 0;
+
+  // operand A: [0, 1] / [1, 2, 0]
+  // operand B: [1, 0] / [2, 1, 0]
+  bool hasBatchDim = srcShape.size() == 3;
+
+  bool matchingOrderA =
+      order.equals({0, 1}) || (hasBatchDim && order.equals({1, 2, 0}));
+  bool matchingOrderB =
+      order.equals({1, 0}) || (hasBatchDim && order.equals({2, 1, 0}));
+  if ((isA && !matchingOrderA) || (!isA && !matchingOrderB))
+    return emitOpError("Order of dimensions don't match expected");
+
+  SmallVector<int64_t> srcShapeBasedOnDstA(dstTy.getShape());
+  srcShapeBasedOnDstA[hasBatchDim ? 1 : 0] /= 2;
+  srcShapeBasedOnDstA[hasBatchDim ? 2 : 1] *= 2;
+
+  SmallVector<int64_t> srcShapeBasedOnDstB(dstTy.getShape());
+  srcShapeBasedOnDstB[hasBatchDim ? 1 : 0] *= 2;
+  srcShapeBasedOnDstB[hasBatchDim ? 2 : 1] /= 2;
+
+  bool aDimMatch = srcShape.equals(ArrayRef(srcShapeBasedOnDstA));
+  bool bDimMatch = srcShape.equals(ArrayRef(srcShapeBasedOnDstB));
+  if ((isA && !aDimMatch) || (!isA && !bDimMatch))
+    return emitOpError(
+        "Input and output dimensions don't match after packing changes");
```

Comment:
I fee it might be more straightforward to read and also less code will run during execution if we just have `if (isA) {...} else {...}` to separate LHS and RHS cases?